java/server/src/org/openqa/grid/internal/RemoteProxy.java

```diff
@@ -39,6 +39,16 @@
  * only support Firefox.
  */
 public interface RemoteProxy extends Comparable<RemoteProxy> {
+
+  /**
+   *
+   * @param protocol - A {@link SeleniumProtocol} object that identifies the request flavor.
+   * @param capabilities - the type of test the client is interested in performing.
+   * @return - The entity on a proxy that can host a test session.
+   */
+  default TestSlot newTestSlot(SeleniumProtocol protocol, Map<String, Object> capabilities) {
```

Comment:
If we resort to a getTestSlotClass() wherein we are returning the actual test slot class, I thought that would bring in reflection (because now the proxy would have to instantiate the class) and we also would need to add additional checks which figures out if the class that was provided satisfies the `isAssignableFrom()` clause and all that. That was why I thought I would keep it simple by just defining a default method in the `RemoteProxy` interface, and any one specifically looking for customizing TestSlots can choose to explicitly override that method from their `DefaultRemoteProxy` extension. But please do let me know if there are other options that can be considered so that I can explore that and get this sorted out.